core/src/main/kotlin/net/corda/core/flows/NotaryWireFormat.kt

```diff
@@ -54,9 +54,13 @@ data class NotarisationRequestSignature(val digitalSignature: DigitalSignature.W
 /**
  * Container for the transaction and notarisation request signature.
  * This is the payload that gets sent by a client to a notary service for committing the input states of the [transaction].
+ * The optional `signatures` carries the list of the transaction signatures potentially (depending on the notary protocol)
+ * required by the notary.
  */
 @CordaSerializable
-data class NotarisationPayload(val transaction: Any, val requestSignature: NotarisationRequestSignature) {
+data class NotarisationPayload(val transaction: Any,
+                               val requestSignature: NotarisationRequestSignature,
+                               val signatures: List<DigitalSignature.WithKey>? = null) {
```

Comment:
```suggestion
                               val transactionSignatures: List<DigitalSignature.WithKey>? = null) {
```
I would make it clear these are signatures over the transaction to be notarised.